external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -553,9 +567,11 @@ public void fail(Object messageId) {
     public void activate() {
         try {
             subscribeKafkaConsumer();
+            metricsConsumer = new KafkaConsumer<>(kafkaSpoutConfig.getKafkaProps());
```

Comment:
Why do we need a separate consumer for the metric? Also if we really need two consumers, you should create this via the factory. Otherwise you prevent us from stubbing/spying on this consumer.